### DISCUSSION TITLE
Button :pressed Style Selector

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#12959] [Q&A] [answered]
I have following xaml

```xaml
  <Window.Styles>
    <Style Selector="Button:pressed">
      <Setter Property="Background" Value="Red"/>
    </Style>
  </Window.Styles>
  <StackPanel>
    <Button Content="A"/>
  </StackPanel>
```
I expect that when the button is pressed it will turn red, but it does nothing.

What did I do wrong?

Test with **Avalonia 11.0.4** Theme **Fluent**

**Comment by workgroupengineering** [comment]
I resolve with this

```xaml
    <Style Selector="Button:pressed /template/ ContentPresenter">
      <Setter Property="Background" Value="Red"/>
    </Style>
```

it is counterintuitive.

It should not be necessary to know the contents of the theme to override a property.

Can i open issue?

**Reply by maxkatz6** [reply, marked as answer]
No, it's old but expected behavior. There already were attempts to change it, which has broke other things. 

This one should explain with with a link to reverted PR as well
https://docs.avaloniaui.net/docs/styling/troubleshooting#selector-with-a-pseudoclass-doesnt-override-the-default

**Reply by workgroupengineering** [reply]
Thanks, I missed it
